### ISSUE TITLE
roberta race finetune

### ISSUE THREAD

**Issue by zengbo1989** [pytorch/fairseq#1114]
I fine-tune the roberta with race dataset according to the readme using the roberta base model but I find the train accuracy is 0.25-0.30 after training for 3-5 epochs, the hyper-parameter is the same with the README.race.md， does this runs normally? I don't make any change to the fairseq code

**Comment by Quentinzzz** [comment]
I got the same problem, also when fine-tuning CommonsenseQA using certain hyper-parameter setting, I guess the reason is that model doesn't converge with bad initialiazation, so try different hyper-parameters might work(It worked when I run the CommonsenseQA task), just that the problem came up pretty often, maybe this has sth to do with RoBERTa's pretraining? it provide a quite special/strange starting point that doesn't easily fit some downstram tasks(just guessing).

Hope someone can give a more reasonable/logical explanation. Thanks!

**Comment by zengbo1989** [comment]
thank you very much @Quentinzzz ! I will try according to your suggestion

**Comment by nelson-liu** [comment]
I've also been trying to reproduce the RoBERTa RACE result, but have yet to have luck on either the base or the large model (performance never goes significantly above random chance). I'm working off of the large model for now, since that seems like what's expected from the README: https://github.com/pytorch/fairseq/blob/master/examples/roberta/README.race.md#3-fine-tuning-on-race

One thing I noticed: looks like `MAX_EPOCH=5`, but table 10 of [the paper](https://arxiv.org/pdf/1907.11692.pdf) indicates that the correct number is 4? Seems like the hyperparameters match otherwise...

Maybe I just got unlucky with the random seeds I tried (default 1 and 2157), but would it be possible to get confirmation that following the instructions in the README does reproduce the result (or at least does better than random chance)?

Aside: would it be possible to rename this issue to something more informative (e.g., "Reproducing RoBERTa RACE result")? Almost missed this when searching for it...

**Comment by nelson-liu** [comment]
Oh: looks like also the command in the README has `warmup_updates=0`, when the paper has it at 6%...I'll try changing that and seeing if there's any luck.

**Comment by grdtsinghua** [comment]
I met the same problem when I fine-tune RoBERTa on Commonsense QA dataset.

**Comment by jingfeidu** [comment]
Hi. Could you try to use multiple GPUs with smaller update frequency? Such as 2 GPUs with 4 update frequency and 4 GPUs with 2 update frequency. We can reproduce the results with these setups and we are looking into the root reason for single GPU setup.

**Comment by myleott** [comment]
I just tried and am getting decent results following the README exactly. Can you share a training log?

**Comment by nelson-liu** [comment]
I'm training roberta base on 2 GPUs right now (don't have a machine with two gpus that are large enough to fit the large version). I tried training with a single GPU (both base and large), to no avail (following instructions in the README exactly besides changing the batch arguments for gradient accumulation).

here's a training log of a single-gpu run i did with roberta large, where I changed the warmup to 6% (as mentioned above): https://gist.github.com/nelson-liu/1d789cb04eaaf81c30857bd5842b8bfa . If it's helpful, I can also rerun with the setting that's exactly in the README on a single GPU and paste the log.

@myleott: would you mind sharing the training log for the run you kicked off that got decent results?

**Comment by Damcy** [comment]
@nelson-liu I've tried the setting in README with 6% warmup steps in one 16GB V100. The final result is
```
| epoch 005 | valid on 'valid' subset | loss 1.107 | nll_loss 0.003 | ppl 1.00 | num_updates 27309 | best_accuracy 0.848987 | accuracy 0.8363
| saved checkpoint checkpoints/checkpoint5.pt (epoch 5 @ 27309 updates) (writing took 38.60006332397461 seconds)
| done training in 111892.1 seconds
```

**Comment by nelson-liu** [comment]
@Damcy thanks for the hint, the model is still training but I do seem to be getting non-random results now. Instead of setting `UPDATE_FREQ=16` and leaving `MAX_SENTENCES=1`, I had set `UPDATE_FREQ=8` and `MAX_SENTENCES=2`---I guess there are probably things within the code that are sensitive to per-GPU batch size, and that it should stay the same? Or should the two setups above be functionally identical?

**Comment by lematt1991** [comment]
Yes, UPDATE_FREQ=16/MAX_SENTENCES=1 has the same effective batch size as UPDATE_FREQ=8/MAX_SENTENCES=2.  I'm going to close this issue as this seems to be resolved.  Please open a new one if you are still having issues.  Thanks!

**Comment by yuweihao** [comment]
What about the weight decay? I am confused that in README, the weight decay is 0.01 while in the paper, the weight decay of RACE is 0.1.

**Comment by jingfeidu** [comment]
both of them should work

**Comment by swarnaHub** [comment]
Hi, can someone (who has been able to reproduce the results) share the finetuned RoBERTa-large model for RACE?
